vpr/src/tileable_rr_graph/tileable_chan_details_builder.cpp

```diff
@@ -179,32 +179,137 @@ ChanNodeDetails build_unidir_chan_node_details(const size_t& chan_width,
+            for (size_t itrack = 0; itrack < num_tracks[iseg]; ++itrack) {
+                bool seg_start = false;
+                bool seg_end = false;
+                /* Every first track of a group of Length-N wires, we set a starting point */
+                if (0 == itrack % seg_len) {
+                    seg_start = true;
+                }
+                /* Every last track of a group of Length-N wires or this is the last track in this group, we set an ending point */
+                if ((seg_len - 1 == itrack % seg_len)
+                    || (itrack == num_tracks[iseg] - 1)) {
+                    seg_end = true;
+                }
+                /* Since this is a unidirectional routing architecture,
+                 * Add a pair of tracks, 1 INC track and 1 DEC track
+                 */
+                int seg_index = segment_inf[iseg].seg_index;
+                chan_node_details.add_track(cur_track, Direction::INC, seg_index, seg_len, seg_start, seg_end);
+                cur_track++;
+                chan_node_details.add_track(cur_track, Direction::DEC, seg_index, seg_len, seg_start, seg_end);
+                cur_track++;
+            }
         }
-        for (size_t itrack = 0; itrack < num_tracks[iseg]; ++itrack) {
-            bool seg_start = false;
-            bool seg_end = false;
-            /* Every first track of a group of Length-N wires, we set a starting point */
-            if (0 == itrack % seg_len) {
-                seg_start = true;
+        else{  // bend segment
+            bend_num++;
+            VTR_ASSERT(segment_inf[iseg].isbend);
+            std::vector<int> seg_len = segment_inf[iseg].part_len;
+            std::vector<int> bend = segment_inf[iseg].bend;
+            VTR_ASSERT(seg_len.size() == 2);   // Only support one bend position for a segment.
+            
+            std::vector<size_t> num_tracks_bend;
+            /* Each bend part tracks number                                               *
+            * For example, a length-5 segment with bend pattern: <- - U -> has 20 tracks. *
+            * Its num_tracks_bend is [20 * 3/5, 20 * 2/5] = [12, 8]                       */
+            for(size_t i = 0; i < seg_len.size(); i++)
+            	num_tracks_bend.push_back(num_tracks[iseg] * seg_len[i] / segment_inf[iseg].length);
+            	
+            VTR_ASSERT(num_tracks_bend[0] + num_tracks_bend[1] == num_tracks[iseg]);
+
+            for (size_t itrack = 0; itrack < num_tracks[iseg]; ++itrack) {
+                bool seg_start = false;
+                bool seg_end = false;
+                size_t seg_bend_start = 0; // seg_bend_start = 0 means not a bend start. 
+                                           // seg_bend_start = i (i > 0) means a bend start for bend segment i.
+                size_t seg_bend_end = 0;   // The same as seg_bend_start.
+                                           // Tracks has same seg_bend_start and seg_bend_end values will be 
```

Comment:
If the tracks in the same ``seg_bend_start`` and ``seg_bend_end`` will be connected. How are the ``seg_bend_start`` and ``seg_bend_end`` look like in a tile or a number of tiles? Any uniqueness is guaranteed?